src/components/vr-scatter-plot/vr-scatter-plot.component.ts

```diff
@@ -1,32 +1,95 @@
-import { Component, ElementRef, Input, OnChanges, OnDestroy, OnInit, SimpleChanges, ViewChild } from '@angular/core';
+import { Component, Input, OnChanges, OnDestroy, OnInit, SimpleChanges } from '@angular/core';
 import { Scatterplot } from '../../d3/scatterplot.d3';
-import 'aframe';
-
-
+import { PreferenceService } from '../../services/preference/preference.service';
+import { DataService } from '../../services/data/data.service';
+import { Meta } from '../../datasets/metas/types';
+import { TimeSeriesQueryOptions } from '../../datasets/queries/time-series.query';
+import { LineChartDatum} from '../line-chart/line-chart.component';
+import { LineChartMeta } from '../../datasets/metas/line-chart.meta';
+import { BehaviorSubject, Subject } from 'rxjs';
+import { DAY } from '../../utils/timeUnits';
+import { map, takeUntil } from 'rxjs/operators';
 
 @Component({
   selector: 'app-vr-scatter-plot',
   templateUrl: './vr-scatter-plot.component.html'
 })
-export class VRScatterPlotComponent implements OnInit, OnChanges, OnDestroy{
+export class VRScatterPlotComponent<T extends Meta>implements OnInit, OnChanges, OnDestroy{
```

Comment:
Add test for this component.

Reply:
just added testing in the most in https://github.com/googleinterns/guide-doge/pull/56/commits/63d9a7efca75808c621de8cb82b664dfe44b0fe6